searchobject/src/main/java/org/jalau/at18/searchobject/middleware/FaceDetectionControllerMiddleware.java

```diff
@@ -29,35 +36,72 @@ public class FaceDetectionControllerMiddleware implements Filter {
-     * @param //request it's the image that user will upload to analyze
-     * @param //response the type of face that we want to detect
+     * @param //request the user request data (all incoming information).
+     * @param  // response the servlet response to the request
      */
     @Override
     public void doFilter(ServletRequest request,
                          ServletResponse response,
                          FilterChain chain) throws IOException, ServletException
     {
         HttpServletRequest req = (HttpServletRequest) request;
+        HttpServletResponse res = (HttpServletResponse) response;
+
+        String path = System.getProperty("user.dir") + "\\src\\main\\java\\org\\jalau\\at18\\searchobject\\middleware\\token\\token.txt";
+        File fileToken = new File(path);
+        Scanner myReader = new Scanner(fileToken);
+        String token = myReader.nextLine(); //Read the txt file
+        int tokenCounter = Integer.parseInt(myReader.nextLine()); //Read the txt file
 
         try{
             LOG.info(" MODEL FACE DETECTION  ");
-            //Verify that an empty or null file isn't entered
-            if(req.getPart("file").getSize() != 0L && req.getPart("file").getSize() > 100 && req.getPart("file").getContentType() != null) {
-                LOG.info(" ACCEPT THE IMAGE ");
-                //Verify that a field isn't empty
-                if(!req.getParameter("type").isEmpty()) {
-                    LOG.info("FACE DETECTION ---- RUNNING -------");
-                    chain.doFilter(request, response);
-                } else {
-                    LOG.warning(" THE TYPE FIELD IS EMPTY ");
+
+            //Verity if have a valid token
+            if (req.getHeader("Authorization").contains(token) && tokenCounter >= 1) {
+                FileWriter fw = new FileWriter(path, false);
+                tokenCounter -= 1; //If it reaches 0 the file is cleaned
+                LOG.info("Remaining token uses: " + String.valueOf(tokenCounter));
+                fw.write(token);
+                fw.write(System.getProperty("line.separator"));
+                fw.write(Integer.toString(tokenCounter));
+                fw.close();
+                myReader.close();
+
+                    //Verify that an empty or null file isn't entered
+                if(req.getPart("file").getSize() != 0L && req.getPart("file").getSize() > 100 && req.getPart("file").getContentType() != null) {
+                    LOG.info(" ACCEPT THE IMAGE ");
+                    //Verify that a field isn't empty
+                    if(!req.getParameter("type").isEmpty()) {
+                        LOG.info("FACE DETECTION ---- RUNNING -------");
+                        chain.doFilter(request, response);
+                    } else {
+                        LOG.warning(" THE TYPE FIELD IS EMPTY ");
+                        throw new MiddlewareException("    The type field is empty ");
```

Comment:
the status error should not also appear here?

Reply:
no, the status error is sent in the catch